lightning/restore/restore.go

```diff
@@ -238,35 +241,11 @@ func NewRestoreControllerWithPauser(
 	return rc, nil
 }
 
-func OpenCheckpointsDB(ctx context.Context, cfg *config.Config) (CheckpointsDB, error) {
-	if !cfg.Checkpoint.Enable {
-		return NewNullCheckpointsDB(), nil
-	}
-
-	switch cfg.Checkpoint.Driver {
-	case config.CheckpointDriverMySQL:
-		db, err := sql.Open("mysql", cfg.Checkpoint.DSN)
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
-		cpdb, err := NewMySQLCheckpointsDB(ctx, db, cfg.Checkpoint.Schema, cfg.TaskID)
-		if err != nil {
-			db.Close()
-			return nil, errors.Trace(err)
-		}
-		return cpdb, nil
-
-	case config.CheckpointDriverFile:
-		return NewFileCheckpointsDB(cfg.Checkpoint.DSN), nil
-
-	default:
-		return nil, errors.Errorf("Unknown checkpoint driver %s", cfg.Checkpoint.Driver)
-	}
-}
-
 func (rc *RestoreController) Close() {
 	rc.backend.Close()
-	rc.tidbMgr.Close()
+	if rc.tidbGlue.OwnsSQLExecutor() {
```

Comment:
why not implement a `Glue.Close` to avoid doing the `own` check?

Reply:
https://github.com/pingcap/tidb-lightning/pull/456/commits/832f65cf8877c1b9228721f86bb77e177a786de8